atd-vze/src/queries/fatalities.js

```diff
@@ -0,0 +1,32 @@
+import { gql } from "apollo-boost";
+
+export const UPDATE_FATALITY = gql`
+  mutation UpdateFatalities(
+    $personId: Int
+    $crashId: Int
+    $changes: fatalities_set_input
+  ) {
+    update_fatalities(
+      where: {
+        crash_id: { _eq: $crashId }
+        _and: {
+          _or: [
+            { person_id: { _eq: $personId } }
+            { primaryperson_id: { _eq: $personId } }
```

Comment:
hmm i think its super unlikely... these ids are default calculated as next val and the most recent `person_id` is 1473645 and `primaryperson_id` is 3955101 so thats a huge range (millions) before they start getting close enough to fear them overlapping. i also did a query of the database now and there are no crashes where the person/primaryperson ids are the same. looking back to the first crashes in our db it looks like the primary person and person values were started with a big gap between them to begin with.

Reply:
I double checked, and the `person_id` and the `primaryperson_id` in their respective tables are surrogate keys which we generate on insert into the table. They do not come from CRIS, and they are only constrained such that they are unique in their respective table. All that to say, they are vanilla `id serial primary key` primary keys. They can, and do in some cases, have potential to overlap. 

The rule of thumb that I use is that the primary key column of every table be nothing except an integer that uniquely identifies the row, and equally importantly, has no imbued meaning encoded in the number. If you can tell something about the data in the row by looking at the ID -- anything at all -- there becomes an opportunity for snags to pop up.

I think that you need to pass some sort of indication along with the `$personId` value about which table you're intending to be querying from along with just the `$personId`. As this is written, if we had an ETL get away from us one day and consume a few million IDs, we'd have to be extra careful in how we cleaned that up because we might loose our method of distinguishing which table's ID field we're intending to match against. And there would be nothing except our own experience telling us that we needed to keep that in mind too, if that were to happen.

I agree with Rose that there is little potential for this to snag us up here, right now, because we just, by luck, have our sequences way out of pace with each other, but I would strongly support spinning off an issue to address this if we don't in this PR.

Reply:
john and i synced yesterday on how to address the possibility of there being a crash record with a primary person and person with the same IDs so ill be pushing that change up today, thanks for the feedback yall!